src/token/Staking/StakingMod.sol

```diff
@@ -0,0 +1,297 @@
+// SPDX-License-Identifier: MIT
+pragma solidity >=0.8.30;
+
+/**
+ * @title Staking Library for Compose
+ * @notice Provides internal logic for staking functionality using diamond storage.
+ *        This library is intended to be used by custom facets to integrate with staking features.
+ * @dev Uses ERC-8042 for storage location standardization.
+ */
+
+/**
+ * @notice Thrown when attempting to stake an unsupported token type.
+ * @param tokenAddress The address of the unsupported token.
+ */
+error StakingUnsupportedToken(address tokenAddress);
+
+/**
+ * @notice Thrown when attempting to stake a zero amount.
+ */
+error StakingZeroStakeAmount();
+
+/**
+ * @notice Thrown when an overflow occurs during arithmetic operations.
+ */
+error StakingOverflow();
+
+/**
+ * @notice Emitted when supported token types are added.
+ */
+event SupportedTokenAdded(address indexed tokenAddress, bool isERC20, bool isERC721, bool isERC1155);
+
+/**
+ * @dev Storage position constant defined via keccak256 hash of diamond storage identifier.
+ */
+bytes32 constant STAKING_STORAGE_POSITION = keccak256("compose.staking");
+
+/**
+ * @notice Structure containing staking information for a specific token.
+ * @param amount The amount of tokens staked.
+ * @param stakedAt The timestamp when the tokens were staked.
+ * @param lastClaimedAt The timestamp when rewards were last claimed.
+ * @param accumulatedRewards The total accumulated rewards for the staked tokens.
+ */
+struct StakedTokenInfo {
+    uint256 amount;
+    uint256 stakedAt;
+    uint256 lastClaimedAt;
```

Comment:
Use `uint64` for timestamps that are stored in contract storage. That will use less gas for writing and reading when they are packed together.